### ISSUE TITLE
Ensure the stage term is mandatory, at least for insect records

### ISSUE THREAD

**Issue by kitenetter** [NERC-CEH/irecord-app#267]
Most of the requests for this come from verifiers of butterfly and moth records, although it also applies to other insect groups. When adding individual records it is currently possible to submit a record without filling in a stage term. The stage term should be mandatory (although it can be set to "not recorded").

The stage term is already mandatory for the moth survey mode, but not for individual records.

On the main forms on the iRecord website we have made the stage term mandatory for all taxon groups (not just insects), and I don't see a problem with doing likewise in the app.

**Comment by kazlauskis** [comment]
We can mandate the stage field, but with the current setup we can't do this for the whole species list, only per species group. Various species groups already have attributes that are based on species-specific iRecord forms, like bryophytes and dragonflies, which have their own stage-specific attributes (e.g. "exuviae-count", "bulbil-true/false"). We could remove these and add a single mandatory stage field, though.

Unlike the website, the app doesn't contain the full taxa hierarchy, it's only a flat list of species with an [informal group ID](https://github.com/NERC-CEH/irecord-app/blob/40d8e303453cdfbe30883d84c9b5dc8e49c3289b/src/common/data/informalGroups.js#L4) attached. The app's dynamic species attributes that are based on informal groups so far were OK, but I feel that this setup is[ pushed to its limits when a new attribute needs to be added to species from different groups.

To re-build a new species dictionary with a different structure and rework how the dynamic attributes work so that we could add attributes for certain branches of the taxonomic hierarchy will be a substantial task. So maybe, for now, we can continue doing this per species group. What do you think? Can you specify which groups we should mandate the stage field?

This is relevant to this [issue](https://github.com/NERC-CEH/irecord-app/issues/211).

**Comment by kitenetter** [comment]
Yes, I can suggest an approach based on the informal group IDs. Will get back to you.

Question: if we do this for iRecord app, will the change be made on LERC Wales and NatureSpot as well? (I hope the answer is yes.)

**Comment by DavidRoy** [comment]
I think not. We don’t have control over external systems? Although you could request?

**Comment by kazlauskis** [comment]
The LERC App is in sync with the iRecord App, so the new stage attribute would be required on both after the changes.

**Comment by kitenetter** [comment]
@kazlauskis following more feedback from moth verifiers in particular, can we prioritise making stage terms mandatory for these informal groups:
- insect - butterfly
- insect - moth

Butterflies already have a taxon-group-specific termlist for stage, which is fine.

Ideally for moths they should use the same taxon-group-specific termlist for stage that is currently available when using the "moth survey" option.
